src/app-layer-detect-proto.c

```diff
@@ -1431,7 +1431,7 @@ AppProto AppLayerProtoDetectGetProto(AppLayerProtoDetectThreadCtx *tctx, Flow *f
         }
     }
 
-    if (!FLOW_IS_PP_DONE(f, flags)) {
+    if (!FLOW_IS_PP_DONE(f, flags) && buf != NULL) {
```

Comment:
would this break probing parsers that only use the port?

Reply:
I think this is something we told ppl that is supported, so best to keep it supported

Reply:
> I think this is something we told ppl that is supported, so best to keep it supported

This would apply to those with custom parsers?